en/developer/reference/articles/02-workflow/05-notification-templates.markdown

```diff
@@ -0,0 +1,287 @@
+| ${stringUtil}                |          | &#10004;  | [StringUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/StringUtil.html) |
+| ${freeMarkerPortletPreferences} |       | &#10004;  | [TemplatePortletPreferences](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-impl/com/liferay/portal/template/TemplatePortletPreferences.html) |
+| ${dateFormats}              | &#10004;  |           | DESCRIBE |
+| ${dateFormatFactory}         |          | &#10004;  | [DateFormatFactory](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/DateFormatFactory.html) |
+| ${userPortraitURL}           | &#10004; |           | DESCRIBE |
+| ${portal}                    |          | &#10004;  | [Portal](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/Portal.html) |
+| ${commonPermission}          |          | &#10004;  | [CommonPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/CommonPermission.html) |
+| ${userURL}                   | &#10004; |           | DESCRIBE |
+| ${expandoValueLocalService}  |          | &#10004;  | [ExpandoValueLocalService](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/expando/kernel/service/ExpandoValueLocalService.html) |
+| ${entryType}                 | &#10004; |           | The type of entry in the workflow (e.g., Blogs Entry) |
+| ${organizationPermission}    |          | &#10004;  | [OrganizationPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/OrganizationPermission.html) |
+| ${passwordPolicyPermission}  |          | &#10004;  | [PasswordPolicyPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/PasswordPolicyPermission.html) |
+| ${expandoColumnLocalService} |          | &#10004;  | [ExpandoColumnLocalService](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/expando/kernel/service/ExpandoColumnLocalService.html) |
+| ${taskComments}              | &#10004; |           | The comments, if any, left by the reviewer |
+| ${staticFieldGetter}         |          | &#10004;  | [StaticFieldGetter](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/StaticFieldGetter.html) |
+| ${htmlUtil}                  |          | &#10004;  | [HtmlUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/HtmlUtil.html) |
+| ${languageUtil}              |          | &#10004;  | [LanguageUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/language/LanguageUtil.html) |
+| ${enumUtil}                  | &#10004; |           | DESCRIBE |
```

Comment:
@rafaprax another one i could use help with :point_up_2:

Reply:
This one is an instance of [TemplateHashModel](https://freemarker.apache.org/docs/api/freemarker/template/TemplateHashModel.html) from freemarker. We put it into freemarker context [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/FreeMarkerTemplateContextHelper.java#L149). For this one, we can point to the freemarker documentation, see https://freemarker.apache.org/docs/pgui_misc_beanwrapper.html#jdk_15_enums.